src/advanced_inv.cpp

```diff
@@ -361,13 +362,22 @@ void advanced_inventory::print_items( advanced_inventory_pane &pane, bool active
         mvwprintz( window, 6 + x, weight_startpos, print_color, "%5.*f", w_precision, it_weight );
 
         //print volume column
-        int it_vol = sitem.volume / units::legacy_volume_factor;
+        double it_vol = units::to_liter( sitem.volume );
+        size_t v_precision;
         print_color = ( it_vol > 0 ) ? thiscolor : thiscolordark;
-        if( it_vol > 9999 ) {
-            it_vol = 9999;
-            print_color = selected ? hilite( c_red ) : c_red;
+        if( it_vol >= 100 ) {
+            if( it_vol >= 10000 ) {
+                it_vol = 9999;
+                print_color = selected ? hilite( c_red ) : c_red;
+            }
+            v_precision = 0;
+        } else if( it_vol >= 10 ) {
```

Comment:
My thinking behind this change is: the `vol` column only has 4 digits worth of space .. if we hardcode `%4.2f` , all volumes `>= 99.99` will make this column overflow to the left. 

So I had to make this code work [the same as for the weight](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/advanced_inv.cpp#L350): dynamic precision, and thus the weight shows the exact same problem you mention, see this capture from the latest master (without this PR):
![cataclysm__dark_days_ahead_-_0_c-18571-g723cee4](https://cloud.githubusercontent.com/assets/1076726/18677968/9a5024d4-7f28-11e6-9c32-d1a22f15a935.png)

Unless there is an option I overlooked, you have to choose between:
- hardcoded precision (2 decimals), for better readability, but may overflow to the left and display useless `0` decimals
- dynamic precision for a better display precision and for consistency with the weight col I guess, but with the decimal point alignment problem you mention

Its up to you.